spring-5-security/src/main/java/com/baeldung/xss/XSSFilter.java

```diff
@@ -0,0 +1,50 @@
+package com.baeldung.xss;
+
+import org.apache.commons.io.IOUtils;
+import org.json.JSONObject;
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import javax.servlet.Filter;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.ServletException;
+import javax.servlet.FilterChain;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+@Order(Ordered.HIGHEST_PRECEDENCE)
+public class XSSFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+        throws IOException, ServletException {
+
+        XSSRequestWrapper wrappedRequest = new XSSRequestWrapper((HttpServletRequest) request);
+
+        String body = IOUtils.toString(wrappedRequest.getReader());
+        if (!"".equals(body)) {
+            JSONObject oldJsonObject = new JSONObject(body);
+            JSONObject newJsonObject = new JSONObject();
+            for (String key : oldJsonObject.keySet()) {
```

Comment:
This function doesn't look like it should be here. We have a `stripXSS` function to manipulate an individual value, surely there should be a helper which iterates over a whole JSON object to strip XSS from its fields?

The other concern here is that this only strips XSS from first-level fields, and it assumes that the object doesn't have arrays or child objects.

Reply:
We need it here before sending the request to the requestWrapper. Otherwise, we lose the request body.

Reply:
yes. because it matches the Person class in this article.